src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -898,9 +898,9 @@ public override AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax
             {
                 throw new ArgumentException("node.Kind==" + node.Kind());
             }
-
-            var bound = GetUpperBoundNode(node);
-            BoundAwaitableInfo awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            var boundAwait = FindBoundAwait(GetLowerBoundNode(node), node)
```

Comment:
We have a helper method for getting all the bound nodes of an expression: `GetBoundNodes`.

Reply:
I'm also not sure that we need to search both upper and lower bound nodes. I would think the upper bound node should be sufficient.

Reply:
Before it was just the upper node - but it was wrapped in a `BoundConversion` which was not handled.
Isn't the lower node just the node we look for?

Reply:
`GetBoundNodes` would be a good fallback if we didn't find it directly? But are there other cases we miss here?

Reply:
Generally, here's how the upper/lower set works: for every syntax node, there is a BoundNode that is highest in the tree, and a bound node that is lowest in the tree. It's very possible that just `GetLowerBoundNode` here will give us the bound node we're interested in with no need to go to the highest node, but that needs to be tested. If not, what I was suggesting is using the `GetBoundNodes` overload that has multiple out parameters for the highest and lowest nodes.